lua/example_of_dynamic_config_request.lua

```diff
@@ -43,6 +44,12 @@ if ret < 0 then
 	return "gk: failed to add an FIB entry\n"
 end
 
+ret = dylib.list_fib_entries(dyc.gk, dylib.c.LIST_LPM_TABLES,
+	"print_fib_dump_entry", nil)
```

Comment:
Instead of passing a string with the name of the handler function, pass the function itself, that is, remove the quotes here and add the prefix `dylib.`. This will avoid the need to look up the function in C.

This also seem to be the bug your facing because `print_fib_dump_entry` is define in `dylib` and being assumed here that it's available as a global variable (i.e. at the same level of the name `dylib`).

Reply:
Good point! I am trying to implement it in this way. The problem is: if we pass the Lua function ``print_fib_dump_entry`` directly, then we need to call it multiple times (depending on the number of dump entries) in ``list_ipv4_fib_entries()`` and ``list_ipv4_fib_entries()``. I wonder how can we keep the parameters (i.e., ``print_fib_dump_entry`` and ``arg``) for multiple calls of ``print_fib_dump_entry``, because after calling ``print_fib_dump_entry``, the lua stack will pop both parameters, and they are not available for next call of ``print_fib_dump_entry``.

Reply:
Copy the value that holds `print_fib_dump_entry()` using `lua_pushvalue()` while setting up the Lua stack to make the call.